corehq/apps/email/forms.py

```diff
@@ -0,0 +1,120 @@
+import re
+
+from crispy_forms import bootstrap as twbscrispy
+from crispy_forms import layout as crispy
+from django import forms
+from django.utils.functional import cached_property
+from django.utils.translation import gettext as _
+
+from corehq.apps.email.models import EmailSettings
+from corehq.apps.hqwebapp import crispy as hqcrispy
+from corehq.motech.const import PASSWORD_PLACEHOLDER
+
+
+class EmailSMTPSettingsForm(forms.ModelForm):
+    username = forms.CharField(required=True)
+
+    plaintext_password = forms.CharField(
+        label=_("Password"),
+        required=True,
+        widget=forms.PasswordInput(render_value=True),
+    )
+
+    server = forms.URLField(
+        label=_('Server'),
+        required=True,
+        help_text=_('e.g. "https://smtp.example.com"'),
+    )
+
+    port = forms.IntegerField(
+        label=_("Port"),
+        required=True,
+        min_value=1,
+        max_value=65535,
+    )
+
+    from_email = forms.EmailField(
+        label=_("Sender's email"),
+        required=True,
+    )
+
+    use_this_gateway = forms.BooleanField(
+        label=_("Use Gateway?"),
+        required=False,
+        help_text=_("Select this option to use this email gateway for sending emails")
+    )
+
+    use_tracking_headers = forms.BooleanField(
+        label=_("Use Tracking Headers"),
+        required=False,
+        help_text=_("Applicable for Amazon's gateway. When selected, "
+                    "emails will be sent with tracking headers along with other headers.")
+    )
+
+    sns_secret = forms.CharField(
```

Comment:
What should be the generation criteria for this field? Is `log_email_event` endpoint an existing one?

Reply:
I'm not sure - I asked on slack https://dimagi.slack.com/archives/C0B1Z40DA/p1695910614809479 but haven't gotten a response yet. I saw you did something in this commit [0cb352b](https://github.com/dimagi/commcare-hq/pull/33518/commits/0cb352b4aa40b915dc7045ee532a227741b464a2) but I'm not following, is the secret generated just 'secret'?

Reply:
Let's ask around again, it'll be best if we can just implement that last piece now instead of having to revert changes you've already made.

Otherwise, if we want to implement this later as a subtask,  we'll need to hide the UI side, and remove it from the model. We can't keep the value as 'secret', otherwise it'll be exposed.
BTW this is where it gets used in my PR.
https://github.com/dimagi/commcare-hq/blob/0a661e033490b5ffd04b35ec8053665fdd1b397f/corehq/apps/hqwebapp/views.py#L1352

Reply:
From Danny's response, we can use https://docs.python.org/3/library/secrets.html `token_urlsafe`

Reply:
Thanks for following up on this. Have added the change. https://github.com/dimagi/commcare-hq/pull/33518/commits/4de9290b9b91e93b424a1915a84918c756011613